core/sessions/oidcauth/oidc.go

```diff
@@ -0,0 +1,675 @@
+/*
+The OIDC module handles authentication by redirecting to a
+Open ID Connect Identity Provider.
+*/
+package oidcauth
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/subtle"
+	"database/sql"
+
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+	"github.com/gin-contrib/sessions"
+	"github.com/gin-gonic/gin"
+	"golang.org/x/oauth2"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/sqlutil"
+	"github.com/smartcontractkit/chainlink-common/pkg/utils/mathutil"
+	"github.com/smartcontractkit/chainlink/v2/core/auth"
+	"github.com/smartcontractkit/chainlink/v2/core/bridges"
+	"github.com/smartcontractkit/chainlink/v2/core/config"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/logger/audit"
+	clsessions "github.com/smartcontractkit/chainlink/v2/core/sessions"
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+	webauth "github.com/smartcontractkit/chainlink/v2/core/web/auth"
+)
+
+const (
+	RouterRateLimitterPeriod = 1 * time.Minute
+	RouterRateLimitterLimit  = 1000
```

Comment:
Unused? Otherwise, spelling:
```suggestion
	RouterRateLimiterPeriod = 1 * time.Minute
	RouterRateLimiterLimit  = 1000
```
And could use a https://pkg.go.dev/golang.org/x/time/rate#Limiter which has some helpers for specifying the rate and supports bursts.

Reply:
Yes, removed 🙂